components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java

```diff
@@ -23,44 +23,230 @@
+                ownerPropertiesDTOList.add(businessOwnerPropertiesDTO);
+            }
+            businessOwnerDTO.setProperties(ownerPropertiesDTOList);
+        } catch (AppManagementException e) {
+            String errorMessage = "Error while saving Business Owner.";
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        }
+        return Response.ok().entity(businessOwnerDTO).build();
     }
 
     @Override
     public Response administrationBusinessownerBusinessOwnerIdGet(String businessOwnerId, String accept, String ifNoneMatch) {
-        return null;
+        BusinessOwnerDTO businessOwnerDTO = new BusinessOwnerDTO();
+        try {
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            //get policy details related to id
+            BusinessOwner businessOwner = apiProvider.getBusinessOwner(Integer.parseInt(businessOwnerId));
+            if (businessOwner == null) {
+                return RestApiUtil.buildNotFoundException("Business Owner ", businessOwnerId).getResponse();
+            }
+            businessOwnerDTO.setName(businessOwner.getBusinessOwnerName());
+            businessOwnerDTO.setEmail(businessOwner.getBusinessOwnerEmail());
+            businessOwnerDTO.setDescription(businessOwner.getBusinessOwnerDescription());
+            businessOwnerDTO.setSite(businessOwner.getBusinessOwnerSite());
+            businessOwnerDTO.setId(businessOwner.getBusinessOwnerId());
+            List<BusinessOwnerProperty> businessOwnerPropertyList = businessOwner.getBusinessOwnerPropertiesList();
+            List<BusinessOwnerPropertiesDTO> businessOwnerPropertiesDTOList = new ArrayList<>();
+            for (BusinessOwnerProperty businessOwnerProperty : businessOwnerPropertyList) {
+                BusinessOwnerPropertiesDTO businessOwnerPropertiesDTO = new BusinessOwnerPropertiesDTO();
+                businessOwnerPropertiesDTO.setKey(businessOwnerProperty.getPropertyId());
+                businessOwnerPropertiesDTO.setValue(businessOwnerProperty.getPropertyValue());
+                businessOwnerPropertiesDTO.setIsVisible(businessOwnerProperty.isShowingInStore());
+                businessOwnerPropertiesDTOList.add(businessOwnerPropertiesDTO);
+            }
+            businessOwnerDTO.setProperties(businessOwnerPropertiesDTOList);
+
+        } catch (AppManagementException e) {
+            String errorMessage = "Error while retrieving details of business owner Business owner Id : " +
+                    businessOwnerId;
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        }
+        return Response.ok().entity(businessOwnerDTO).build();
     }
 
     @Override
     public Response administrationBusinessownerBusinessOwnerIdPut(String businessOwnerId, BusinessOwnerDTO body, String contentType, String ifMatch, String ifUnmodifiedSince) {
-        return null;
```

Comment:
Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/0d9719587a489bf2ef909c862c46bdbce01f1af0